README.md

```diff
@@ -64,15 +64,18 @@ An opinioned coding guideline for React + Redux project @WORD UP
 ## React
 #### Add `propTypes` to every components
   * It helps others to use the component easily and also help to catch some errors quickly.
-  * In alphabetic order
+  * Order by category
 
 #### Use multiple lines when we need to pass more than one props.
-  * Organize props in alphabetic order.
+  * Organize props by category.
     ```javascript
-    <Component
-      aProp: 'a data'
-      bProp: 'b data'
-    />
+      <Input
+        ref={(ref) => this.BTN = ref}
+        onBlur={this.handleBlur}
+        onChange={this.handleChange}
+        height={height}
+        width={width}
+      />
```

Comment:
Actually, I agree with you. But I just afraid we don't have a good tool to logically define `category`, so in the end, it becomes everyone has his/her own decision. Or do you have any recommendation tools?

Reply:
No, but in my experience, this won't be a problem if we don't apply the same order for props.
For the two examples below are both good for me.
Because I only wish to see all props about style(height/width) or all props about event handle(onChange/onBlur) at the same time.
```
<Input
   ref={(ref) => this.BTN = ref}
   onBlur={this.handleBlur}
   onChange={this.handleChange}
   height={height}
   width={width}
/>
```
```
<Input
   width={width}
   height={height}
   ref={(ref) => this.BTN = ref}
   onChange={this.handleChange}
   onBlur={this.handleBlur}
/>
```

Reply:
ok. got your point. Previous I set this rule since I found it's quite difficult to compare the usage and definitoin like below. So, i think if we all restrict it to be ordered alphabetically. Then the usage and definition will be always in the same order. In your opinion, do you think the consistency between these two out-weight the `meaningfull group`(by category)?

```javascript
<Footer
  disableReview={ !this.hasMoreThanOneCard() }
  location={ this.props.location }
  sortCardDir={ this.props.reducer.get('sortCardDir') }
  deckId={ this.getDeckId() }
  closeFooter={ this.props.actions.closeFooter }
  handleEnterSumbit={ this.handleEnterSumbit }
  handleSearchInput={ this.handleFooterInput }
  handleSort={ this.handleFooterSort }
  footerControlToShow={ this.props.reducer.get('footerControlToShow') }
  showFooterControls={ this.props.reducer.get('showFooterControls') }
  displayFooterControl={ this.props.actions.displayFooterControl }
  clickReview={ this.showReviewSetting }
/>


Footer.propTypes = {
  clickReview: PropTypes.func,
  closeFooter: PropTypes.func.isRequired,
  deckId: PropTypes.string,
  disableReview: PropTypes.bool,
  displayFooterControl: PropTypes.func.isRequired,
  footerControlToShow: PropTypes.string.isRequired,
  handleEnterSumbit: PropTypes.func.isRequired,
  handleSort: PropTypes.func,
  handleSearchInput: PropTypes.func.isRequired,
  location: PropTypes.object,
  sortCardDir: PropTypes.string,
  showFooterControls: PropTypes.bool.isRequired,
};
```

BTW, if you think the `meaningfull group` is still better. Then how about we set the rules that:
1. Order props by category. And add comments or spaces if needed. For example
```javascript
Input.propTypes {
   // styles
   width: PropTypes.string.isRequired,
   height: PropTypes.string.isRequired,
   // event handlers
   onChange: PropTypes.func.isRequired,
   onBlurPropTypes.func.isRequired,

   ref={(ref) => this.BTN = ref}  // there's a space as separator
}
```

2. Follow the order as `propTypes` when you use the component:
```javascript
<Input
   width={width}
   height={height}
   onChange={this.handleChange}
   onBlur={this.handleBlur}
   ref={(ref) => this.BTN = ref}
/>
```

Reply:
1. About the `Footer`, I don't think order props will help us to understand the meaning, the only thing we can know is just the type of the props. I believe adjust naming and try to reduce the props will be better than spend time on order the props every time we use it.
2. With proper naming, I think we won't have to add the comment.